code/RedirectedURLCsvBulkLoader.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+class RedirectedURLCsvBulkLoader extends CsvBulkLoader
+{
+    private $exception_thrown;
+
+    /**
+     * @todo Better messages for relation checks and duplicate detection
+     * Note that columnMap isn't used.
+     * overriding this here for handling ValidationException when DataObject::write() called
+     *
+     * @param array $record
+     * @param array $columnMap
+     * @param BulkLoader_Result $results
+     * @param boolean $preview
+     *
+     * @return int
```

Comment:
Please add extra doc to say this int is the object ID

Reply:
Done